### PR TITLE
feat: gas optimization 

### PR DESCRIPTION
**Pull request by ChefMist**
Refactoring changes which leads to gas optimization

### CODE REVIEW

**Comment by ChefMist** [comment]
ready for review

**Review by chefburger — APPROVED** [review]
lgtm

File: src/base/Dispatcher.sol [hunk 1]
```diff
@@ -209,11 +211,14 @@ abstract contract Dispatcher is
                             permitSingle := inputs.offset
                         }
                         bytes calldata data = inputs.toBytes(6); // PermitSingle takes first 6 slots (0..5)
-                        try PERMIT2.permit(msgSender(), permitSingle, data) {}
-                        catch (bytes memory reason) {
-                            output = reason;
-                            success = false;
-                        }
+                        (success, output) = address(PERMIT2).call(
+                            abi.encodeWithSignature(
```
Comment (chefburger) [inline src/base/Dispatcher.sol:215]: may be a tiny one, does abi.encodeCall better ?

Comment (jpopxfile) [reply src/base/Dispatcher.sol:215]: Yes encodeCall provides type safety.

Comment (ChefMist) [reply src/base/Dispatcher.sol:215]: we can't use that as there's multiple `permit` function

Comment (ChefMist) [reply src/base/Dispatcher.sol:215]: eg. this is the error `Member "permit" not unique after argument-dependent lookup in type(contract IAllowanceTransfer)`

Comment (ChefMist) [reply src/base/Dispatcher.sol:215]: will merge for now, though if there's any follow-up action required, will do follow-up PR to resolve the comment

**Pull request merged**